### PR TITLE
Remove telemetry from authorize URL

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

Remove telemetry from authorize URL. 

### Testing

* [x] This change adds unit test coverage to make sure no telemetry is added

### CODE REVIEW

File: spec/omniauth/strategies/auth0_spec.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (lbalmaceda) [inline spec/omniauth/strategies/auth0_spec.rb]: why did you need to add this?

File: lib/omniauth/strategies/auth0.rb [hunk 2]
```diff
@@ -75,16 +77,14 @@ def client
...
+        parsed_query = Rack::Utils.parse_query(request.query_string)
+        params['connection'] = parsed_query['connection']
+        params['prompt'] = parsed_query['prompt']
         params
       end
 
       def build_access_token
-        telemetry_header = { 'Auth0-Client' => telemetry_encoded }
-        options.token_params.merge!(:headers => telemetry_header)
+        options.token_params[:headers] = { 'Auth0-Client' => telemetry_encoded }
```
Comment (lbalmaceda) [inline lib/omniauth/strategies/auth0.rb]: previously this said "merge". Isn't this now a replace rather than a merge? 🤔

File: spec/omniauth/strategies/auth0_spec.rb (hunk 1 above)
Comment (joshcanhelp) [reply spec/omniauth/strategies/auth0_spec.rb]: `rubocop` suggestion, I don't think it has much effect on what's happening here but idiomatic.

File: lib/omniauth/strategies/auth0.rb (hunk 2 above)
Comment (joshcanhelp) [reply lib/omniauth/strategies/auth0.rb]: Should be doing the same. Previous was merging `options.token_params` with `:headers => telemetry_header`, not merging headers (since `:headers` might not be a hash or even exist).
